src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
+
+  let sum = 0;
+
+  women.forEach((w) => sum += w.died - w.born);
+
+  return sum / women.length;
```

Comment:
Avoid using `forEach`. Here you can use `reduce` method, it will return the sum, so no need to create `sum` variable and store the value there. 

Using `reduce` the implementation could be like this:
```suggestion
 const result = women.reduce((prev, {born, died}) => {
    return (died - born) + prev;
}, 0)'
```

Reply:
Try to avoid naming variables like `m`
It makes it difficult to read the code

Reply:
Also instead of using  `woman.died`, `woman. born` you can use ES6 destructuring:
`{born, died}`